### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by oadiazp**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 *.egg-info/
 .cache/
 __pycache__/
+/.idea/
```
Comment (KonstantinSchubert) [inline .gitignore]: put a new line at the end :)

File: carbon14/django.py [hunk 2]
```diff
@@ -86,3 +86,24 @@ def serialize(self, instance, *args, **kwargs):
         if isinstance(value, Point):
             value = tuple(value)
         return value
+
+
+class PaginatedModelCollection(ModelCollection):
+    def _resolve(
```
Comment (KonstantinSchubert) [inline carbon14/django.py]: I think that we can put this directly into ModelCollection.

Offset and limit can be none by default. 
And only if they are not none, they have an effect.

File: carbon14/django.py [hunk 3]
```diff
@@ -86,3 +86,24 @@ def serialize(self, instance, *args, **kwargs):
...
+            limit=0,
+            **kwargs
+    ):
+        return super()._resolve(
+            level,
+            instances,
+            ctx,
+            ids,
+            **kwargs
+        )[offset:limit]
```
Comment (KonstantinSchubert) [inline carbon14/django.py]: logically, this should be [offset: limit+offset]

File: carbon14/django.py [hunk 4]
```diff
@@ -44,13 +44,34 @@ class ModelCollection(Collection):
...
 
-    def _resolve(self, level, instances, ctx, ids=None, **kwargs):
+    def _resolve(
+            self,
+            level,
+            instances,
+            ctx,
+            ids=None,
+            limit=None,
+            offset=None,
```
Comment (edelvalle) [inline carbon14/django.py]: Make offset by default 0

File: carbon14/django.py [hunk 5]
```diff
@@ -44,13 +44,34 @@ class ModelCollection(Collection):
...
+            **kwargs
+    ):
         if self._auth_required and not ctx.user.is_authenticated():
             instances = instances.none()
         else:
             if ids is not None:
                 instances = instances.filter(id__in=ids)
-        return instances.all()
+
+        instances = instances.all()
```
Comment (edelvalle) [inline carbon14/django.py]: for this section I suggest:

```python
instances = instance.all()[offset:]

if limit:
    instances = instances[:limit]

return instances
```

**Comment by KonstantinSchubert** [comment]
@edelvalle I think with the new changes, this could be merged?

**Pull request merged**